pkg/sync/sync.go

```diff
@@ -36,6 +47,42 @@ type ReplicaRebuildStatus struct {
 	FromReplicaAddress string `json:"fromReplicaAddress"`
 }
 
+func NewTaskError(res ...*ReplicaError) *TaskError {
+	return &TaskError{
+		ReplicaErrors: append([]*ReplicaError{}, res...),
+	}
+}
+
+func (e *TaskError) Error() string {
+	var errs []string
+	for _, re := range e.ReplicaErrors {
+		errs = append(errs, re.Error())
+	}
+
+	if errs == nil {
+		return "Unknown"
```

Comment:
If there is no error, the error object itself should be `nil` rather than containing an empty error message. Hence we shouldn't return/use this `TaskError` if there is no `ReplicaError`.

BTW, the definition of `TaskError` is similar to this `https://github.com/longhorn/longhorn-engine/blob/c10cd31baccd8008c25d5f1befc26196cb6c2622/pkg/controller/replicator.go#L34`